pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/MPSTNCuda.hpp

```diff
@@ -307,6 +337,26 @@ class MPSTNCuda final : public TNCudaBase<Precision, MPSTNCuda<Precision>> {
     }
 
   private:
+    /**
+     * @brief Return bondDims to the member initializer
+     * NOTE: This method only works for the open boundary condition
+     * @return std::vector<std::size_t>
+     */
+    std::vector<std::size_t> setBondDims_() {
+        std::vector<std::size_t> localBondDims(BaseType::getNumQubits() - 1,
+                                               maxBondDim_);
+
+        for (std::size_t i = 0; i < localBondDims.size(); i++) {
```

Comment:
@maliasadi Is that an example of a loop that we would replace by `std::transform`?

Reply:
hmmm, I would prefer `std::transform` only if the operation function is less than `3` lines. How do you think @vincentmr @maliasadi ?

Reply:
I'm favour of using `std::transform` here as well. I feel close to some of the answers [here](https://stackoverflow.com/questions/39174060/when-to-prefer-for-loop-over-stdtransform-or-vice-versa). But it's not a blocker.. look good either ways!